src/class-parsely-recommended-widget.php

```diff
@@ -108,115 +108,31 @@ public function widget( $args, $instance ) {
-				if (img_src !== 'none') {
-					outerDiv.addClass('display-thumbnail');
-				}
-				if (display_direction) {
-					outerDiv.addClass('list-' + display_direction);
-				}
-
-				var outerList = jQuery('<ul>').addClass('parsely-recommended-widget').appendTo(outerDiv);
-				jQuery.getJSON( full_url, function (data) {
-					jQuery.each(data.data, function(key, value) {
-						var widgetEntry = jQuery('<li>')
-							.addClass('parsely-recommended-widget-entry')
-							.attr('id', 'parsely-recommended-widget-item' + key);
-
-						var textDiv = jQuery('<div>').addClass('parsely-text-wrapper');
-
-						if (img_src === 'parsely_thumb') {
-							jQuery('<img>').attr('src', value['thumb_url_medium']).appendTo(widgetEntry);
-						}
-						else if (img_src === 'original') {
-							jQuery('<img>').attr('src', value['image_url']).appendTo(widgetEntry);
-						}
-
-						var cmp_cmp = '?itm_campaign=<?php echo esc_attr( $this->id ); ?>';
-						var cmp_med = '&itm_medium=' + itm_medium;
-						var cmp_src = '&itm_source=' + itm_source;
-						var cmp_con = '&itm_content=widget_item-' + key;
-						var itm_link = value['url'] + cmp_cmp + cmp_med + cmp_src + cmp_con;
-
-						var postTitle = jQuery('<div>').attr('class', 'parsely-recommended-widget-title');
-						var postLink = jQuery('<a>').attr('href', itm_link).text(value['title']);
-						postTitle.append(postLink);
-						textDiv.append(postTitle);
-
-						if ( display_author ) {
-							var authorLink = jQuery('<div>').attr('class', 'parsely-recommended-widget-author').text(value['author']);
-							textDiv.append(authorLink);
-						}
-
-						widgetEntry.append(textDiv);
-
-
-
-						// set up the rest of entry
-						outerList.append(widgetEntry);
-					});
-					outerDiv.append(outerList);
-				});
-
-			}
-			defer(widgetLoad);
+		$reco_widget_script_asset = require PARSELY_PLUGIN_DIR . 'build/admin-page.asset.php';
+
+		wp_register_script(
+			'wp-parsely-recommended-widget',
+			PARSELY_PLUGIN_URL . 'build/recommended-widget.js',
+			$reco_widget_script_asset[ 'dependencies' ],
+			PARSELY::get_asset_cache_buster(),
+			true
+		);
 
+		wp_localize_script(
+			'wp-parsely-recommended-widget',
+			'wpParselyRecommended',
+			array(
+				'displayAuthor' => isset( $instance['display_author'] ) ? wp_json_encode( boolval( $instance['display_author'] ) ) : false,
+				'displayDirection' => isset( $instance['display_direction'] ) ? esc_js( $instance['display_direction'] ) : null,
+				'fullUrl' => esc_js( esc_url_raw( $full_url ) ),
+				'imgSrc' => isset( $instance['img_src'] ) ? esc_js( $instance['img_src'] ) : null,
+				'permalink' => wp_json_encode( esc_url_raw( get_permalink() ) ),
+				'personalized' => wp_json_encode( boolval( $instance['personalize_results'] ) ),
+				'widgetId' => esc_attr( $this->id ),
+			)
+		);
 
-		</script>
-		<?php
+		wp_enqueue_script( 'wp-parsely-recommended-widget' );
 
 		echo wp_kses( $args['after_widget'], $allowed_tags );
```

Comment:
Any reason this can't just be:

```suggestion
		echo wp_kses_post( $args['after_widget'] );
```

Reply:
Considering line 97 is `$allowed_tags = wp_kses_allowed_html( 'post' );`, then it would seem that `wp_kses_post()` is sufficient.

Check line 99 as well.

Reply:
Changed it over in #290 since the change isn't really related to the rest of the spirit of this PR.